### ISSUE TITLE
The reproduce results not good

### ISSUE THREAD

**Issue by dsdfasfaac** [robocasa/robocasa#204]
Hi, thanks for releasing the RoboCasa GR00T benchmark code.

I am trying to reproduce the GR00T N1.5 training results on RoboCasa365. I trained the model using this repository on 8×H100 GPUs with the RoboCasa365 pretrain human demonstration data.

My setup is as follows:

```text
Repository: https://github.com/robocasa-benchmark/Isaac-GR00T
Model: GR00T N1.5
Dataset: RoboCasa365 pretrain data, human demonstrations only
Hardware: 8×NVIDIA H100
Per-GPU batch size: 8
Effective global batch size: 128
Other training configurations: same as the open-source repository
```

The training loss appears to converge normally. In my run, the final loss is around `0.045`, but the closed-loop evaluation success rate is still very low, roughly around `10%`. This is much lower than what I expected from the reported GR00T N1.5 RoboCasa365 results.

For reference, here is my W&B training report:

```text
https://api.wandb.ai/links/2412777093-sdadaad/zanc43s9
```

Any guidance on the exact reproduction configuration would be very helpful.

Thanks.

**Comment by Abhiram824** [comment]
Hi, I will take a look at the training config. In the meantime if you have the resources, please try the following things:

1. Keep global batch size=128, but maximize per device batch size. (likely batch size 64, ngpus=2)
2. Evaluate the public checkpoint: https://huggingface.co/robocasa/robocasa365_checkpoints/tree/main/gr00t_n1-5/multitask_learning/checkpoint-120000 and check that you get the same results as reported in the paper

**Comment by sepnasiriany** [comment]
I already answered your question in the issue you brought up earlier, but the training configuration is in the wandb shared. Please clarify the specific configuration you're looking for that wasn't included there. As @Abhiram824 mentioned, I think it's more helpful that you evaluate our current checkpoint and see if you can get similar results. If you can't, then there may be an issue with your evaluation pipeline, not your training.

**Comment by dsdfasfaac** [comment]
Thanks for the suggestions.

I realized that I made a typo in my previous description. My original training setup was:

```
per-device batch size = 16
num_gpus = 8
effective global batch size = 128
gradient_accumulation_steps = 1
```

So the effective global batch size should indeed be 128.

I have now evaluated the [public checkpoint](https://huggingface.co/robocasa/robocasa365_checkpoints/tree/main/gr00t_n1-5/multitask_learning/checkpoint-120000):

```
robocasa/robocasa365_checkpoints/gr00t_n1-5/multitask_learning/checkpoint-120000
```

So far, I have only evaluated the `atomic_seen` tasks. The average success rate I get is `45.0%`, compared with the `50.7%` reported in the paper. There is still a gap of about 5.7 percentage points, but this is much better than the checkpoint I trained using the public training configuration, which only achieves around `6.2%`.

Here are my results for the public checkpoint on `atomic_seen`:

```
CloseBlenderLid:              0.14
CloseFridge:                  0.70
CloseToasterOvenDoor:         0.38
CoffeeSetupMug:               0.10
NavigateKitchen:              0.06
OpenCabinet:                  0.58
OpenDrawer:                   0.40
OpenStandMixerHead:           0.58
PickPlaceCounterToCabinet:    0.80
PickPlaceCounterToStove:      0.78
PickPlaceDrawerToCounter:     0.50
PickPlaceSinkToCounter:       0.80
PickPlaceToasterToCounter:    0.64
SlideDishwasherRack:          0.58
TurnOffStove:                 0.06
TurnOnElectricKettle:         0.62
TurnOnMicrowave:              0.18
TurnOnSinkFaucet:             0.36

Average:                      0.45
```

Based on the current results, it seems that the main issue is more likely in my training pipeline rather than the evaluation pipeline.

I also try the larger per-device batch setting you suggested (per_batchsize=64, gpus=2). but it is not work.

**Comment by sepnasiriany** [comment]
Can you share the metadata.json (normalization stats) from your trained checkpoint and compare it to the public checkpoint-120000, and confirm the normalization_mode for action/state in your data config?

**Comment by dsdfasfaac** [comment]
I compared the metadata.json [metadata.json](https://github.com/user-attachments/files/29074494/metadata.json) from our trained checkpoint-120000 with the public robocasa365_multitask/checkpoint-120000. The modality schema and all state/action statistic keys and shapes are identical. The files are not byte-identical, but the numeric differences are very small: over 168 scalar statistics, max absolute difference is 0.00693, mean absolute difference is 3.47e-4, and all scalar differences are <= 1e-2.

For our data config, PandaOmronDataConfig uses min_max normalization for all state keys. For actions, end_effector_position, end_effector_rotation, and base_motion use min_max; gripper_close and control_mode use binary.

**Comment by sepnasiriany** [comment]
what command are you running when you run the training with `gr00t_finetune.py`?

**Comment by dsdfasfaac** [comment]
```
.venv/bin/python3 -u scripts/gr00t_finetune.py --dataset-soup pretrain_human300 --output-dir ./outputs --data-config panda_omron --batch-size 16 --max-steps 150000 --num-gpus 8 --save-steps 20000 --base-model-path ./playground/Pretrain_models/GR00T-N1.5-3B
```
torch 2.5.1+cu124
flash_attn 2.7.1.post4

**Comment by sepnasiriany** [comment]
okay this makes sense, you're training with `--data-config panda_omron` but we use `--data-config panda_omron_task_name_lang`. That config feeds the per-task name/language conditioning the policy needs, the plain `panda_omron` trains to the same low loss but isn't conditioned on the task instruction, so it collapses at eval. Try with `panda_omron_task_name_lang` and let us know how it goes

**Comment by dsdfasfaac** [comment]
Thanks for the clarification.

We tried `--data-config panda_omron_task_name_lang` as suggested, but our current results seem a bit different from the expected behavior.

Using the same evaluation setup, we obtained the following results:

```text
Data config: panda_omron

Atomic-Seen:       49.0
Composite-Seen:    12.25
Composite-Unseen:  3.875
```

```text
Data config: panda_omron_task_name_lang

Atomic-Seen:       32.1
Composite-Seen:    2.75
Composite-Unseen:  1.25
```

So in our current experiments, `panda_omron` actually performs better than `panda_omron_task_name_lang`, although both are still not fully matching the released checkpoint / reported results.

We also found another important issue in our previous training pipeline. Since we trained with DDP, the loss was already averaged once, but in our implementation it was averaged again before backpropagation. This effectively changed the loss / gradient scaling compared with the official one-GPU training setup, and likely caused the large discrepancy we observed earlier.

After fixing this DDP loss scaling issue, the performance improved significantly, which suggests that the main issue in our previous failed training run may have been the DDP loss reduction rather than only the data config.

**Comment by sepnasiriany** [comment]
Could you share the exact change you made for the DDP loss fix?
